packages/core/src/odata/common/selectable/enum-field.ts

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+/* eslint-disable max-classes-per-file */
+
+import { EntityBase, Constructable } from '../entity';
+import { ComplexTypeField, getEntityConstructor } from './complex-type-field';
+import { ConstructorOrField } from './constructor-or-field';
+import { EdmTypeField, SelectableEdmTypeField } from './edm-type-field';
+
+/**
+ * Represents a property with a enum value.
+ *
+ * @typeparam EntityT - Type of the entity the field belongs to
+ */
+class EnumFieldBase<EntityT extends EntityBase> extends EdmTypeField<
+  EntityT,
+  string
```

Comment:
Add a  comment here why string is used as a temporary solution to the type problem.